connector_amazon/models/amazon_sale_importer.py

```diff
@@ -3,14 +3,204 @@
+        for line in reader:
+            if not line.get('order-item-id'):
+                continue
+            if line['order-id'] in sales:
+                vals = sales[line['order-id']]
+                vals['lines'].append(self._get_sale_line(line))
+            else:
+                vals = {
+                    'sale': {
+                        'origin': line['order-id'],
+                        'date_order': line['purchase-date'],
+                    },
+                    'partner': {
+                        'email': line['buyer-email'],
+                        'name': line['buyer-name'],
+                        'phone': line['buyer-phone-number'],
+                    },
+                    'part_ship': {
+                        'name': line['recipient-name'],
+                        'phone': reset_empty(line['ship-phone-number']),
+                        'street': reset_empty(line['ship-address-1']),
+                        'street2': reset_empty(line['ship-address-2']),
+                        'street3': reset_empty(line['ship-address-3']),
+                        'city': line['ship-city'],
+                        'state': reset_empty(line['ship-state']),
+                        'zip': reset_empty(line['ship-postal-code']),
+                        'country': line['ship-country'],
+                    },
+                    'lines': [self._get_sale_line(line)],
+                }
+            sales[line['order-id']] = vals
+        return sales
+
+    def _get_sale_line(self, line):
+        return {
+            'item': line['order-item-id'],
+            'sku': line['sku'],
+            'name': '[%s] %s' % (line['sku'], line['product-name']),
+            'product_uom_qty': line['quantity-purchased'],
+            # price is in full tax, vat is computed in odoo
+            'price_unit': float(line['item-price']) + float(line['item-tax']),
+            'shipping': float(line['shipping-price']) + \
+            float(line['shipping-tax']),
+        }
+
+    def _create_sales(self, sales, meta_attachment):
+        """ We process all sale orders of the file
+        """
+        backend = meta_attachment.amazon_backend_id
+        for order_item, sale in sales.items():
+            partner, part_ship = self._get_partners(sale)
+            vals = {
+                'name': '%s %s' % (
+                    backend.sale_prefix or '',
+                    sale['sale']['origin']),
+                'partner_id': partner.id,
+                'partner_shipping_id': part_ship.id,
+                'pricelist_id': backend.pricelist_id.id,
+                'external_origin': 'ir.attachment.metadata,%s'
+                % meta_attachment.id,
+                'origin': sale['sale']['origin'],
+            }
+            if sale.get('lines'):
+                ship_price = self._complete_products(sale['lines'], backend)
+                vals['order_line'] = [
+                    (0, '_', {key: val for key, val in line.items()})
```

Comment:
Nitpic: `'_'` - no need to put here string, more common is to put 0 like a second value.